core/templates/domain/exploration/read-only-exploration-backend-api.service.spec.ts

```diff
@@ -95,139 +72,150 @@ describe('Read only exploration backend API service', function() {
-    }
+      expect(failHandler).toHaveBeenCalled();
+    })
   );
 
-  it('should report caching and support clearing the cache', function() {
-    var successHandler = jasmine.createSpy('success');
-    var failHandler = jasmine.createSpy('fail');
+  it('should report caching and support clearing the cache', fakeAsync(() => {
+    let successHandler = jasmine.createSpy('success');
+    let failHandler = jasmine.createSpy('fail');
 
     // The exploration should not currently be cached.
-    expect(ReadOnlyExplorationBackendApiService.isCached('0')).toBe(false);
+    expect(readOnlyExplorationBackendApiService.isCached('0')).toBe(false);
 
     // Loading a exploration the first time should fetch it from the backend.
-    $httpBackend.expect('GET', '/explorehandler/init/0').respond(
-      sampleDataResults);
-    ReadOnlyExplorationBackendApiService.loadLatestExploration('0').then(
+    readOnlyExplorationBackendApiService.loadLatestExploration('0').then(
       successHandler, failHandler);
-    $httpBackend.flush();
+    var req = httpTestingController.expectOne(
+      '/explorehandler/init/0');
+    expect(req.request.method).toEqual('GET');
+    req.flush(sampleDataResults);
 
-    expect(successHandler).toHaveBeenCalledWith(sampleDataResults);
+    flushMicrotasks();
+
+    expect(successHandler).toHaveBeenCalled();
     expect(failHandler).not.toHaveBeenCalled();
 
     // The exploration should now be cached.
-    expect(ReadOnlyExplorationBackendApiService.isCached('0')).toBe(true);
+    expect(readOnlyExplorationBackendApiService.isCached('0')).toBe(true);
 
     // The exploration should be loadable from the cache.
-    ReadOnlyExplorationBackendApiService.loadLatestExploration('0').then(
+    readOnlyExplorationBackendApiService.loadLatestExploration('0').then(
       successHandler, failHandler);
     expect(successHandler).toHaveBeenCalledWith(sampleDataResults);
     expect(failHandler).not.toHaveBeenCalled();
 
     // Resetting the cache will cause another fetch from the backend.
-    ReadOnlyExplorationBackendApiService.clearExplorationCache();
-    expect(ReadOnlyExplorationBackendApiService.isCached('0')).toBe(false);
+    readOnlyExplorationBackendApiService.clearExplorationCache();
+    expect(readOnlyExplorationBackendApiService.isCached('0')).toBe(false);
 
-    $httpBackend.expect('GET', '/explorehandler/init/0').respond(
-      sampleDataResults);
-    ReadOnlyExplorationBackendApiService.loadLatestExploration('0').then(
+    readOnlyExplorationBackendApiService.loadLatestExploration('0').then(
       successHandler, failHandler);
-    $httpBackend.flush();
+    var req = httpTestingController.expectOne(
+      '/explorehandler/init/0');
+    expect(req.request.method).toEqual('GET');
+    req.flush(sampleDataResults);
 
-    expect(successHandler).toHaveBeenCalledWith(sampleDataResults);
+    flushMicrotasks();
+
+    expect(successHandler).toHaveBeenCalled();
```

Comment:
ditto. Please make this change here and other places in this file.

Reply:
Done